src/sentry/static/sentry/app/components/grid.jsx

```diff
@@ -0,0 +1,217 @@
+/* eslint-disable no-unused-vars */
+/* eslint-disable react/no-unused-prop-types */
+
+// This modules is intended to be a drop-in replacement for grid-emotion
+// with a similar API but is based on our app's spacing and breakpoints
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'react-emotion';
+import space from '../styles/space';
+import theme from '../utils/theme';
+
+class Base extends React.Component {
+  static propTypes = {
+    w: PropTypes.oneOfType([PropTypes.number, PropTypes.array]),
+    m: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+    mx: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+    my: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+    ml: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+    mr: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+    mb: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+    mt: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+    p: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+    px: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+    py: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+    pl: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+    pr: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+    pb: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+    pt: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+    flex: PropTypes.string,
+    alignItems: PropTypes.string,
+    justifyContent: PropTypes.string,
+    flexDirection: PropTypes.string,
+    flexWrap: PropTypes.string,
+  };
+
+  getSpacingRules = () => {
+    const spacingProps = [
+      ['m', ['margin']],
+      ['mx', ['margin-left', 'margin-right']],
+      ['my', ['margin-top', 'margin-bottom']],
+      ['ml', ['margin-left']],
+      ['mr', ['margin-right']],
+      ['mt', ['margin-top']],
+      ['mb', ['margin-bottom']],
+      ['p', ['padding']],
+      ['px', ['padding-left', 'padding-right']],
+      ['py', ['padding-top', 'padding-bottom']],
+      ['pl', ['padding-left']],
+      ['pr', ['margin-right']],
+      ['pt', ['padding-top']],
+      ['pb', ['padding-bottom']],
+    ];
+
+    const rules = [];
+
+    spacingProps.forEach(([prop, cssProperties]) => {
+      const val = this.getSpacingValue(this.props[prop]);
+      if (typeof val !== 'undefined') {
+        cssProperties.forEach(property => {
+          rules.push(`${property}:${val};`);
+        });
+      }
+    });
+
+    return rules;
+  };
+
+  getSpacingValue = prop => {
+    switch (typeof prop) {
+      case 'number':
+        return space(prop);
+      case 'string':
```

Comment:
fwiw i would be fine ignoring string values. We haven't used one yet (!!) and it seems like bad style anyways.